bin/hdfcoinc/pycbc_coinc_statmap

```diff
@@ -19,112 +22,188 @@ class fw(object):
+ave_fore_time = (fore_time1 + fore_time2) / 2.0
+
+# remove start and end time around every average foreground trigger time to
+# window around.
+remove_start_time = ave_fore_time - args.veto_window
+remove_end_time = ave_fore_time + args.veto_window
+
+# The time contained between segments around the times contained
+# between each element of remove_start_time and remove_end_time
+
+veto_time = abs(veto.start_end_to_segments(remove_start_time,
+                                           remove_end_time).coalesce())
+
+# Veto indices from list of triggers for times in ifo 1 around the 
+# window times
+veto_indices1 = veto.indices_within_times(all_trigs.time1, remove_start_time,
+                                          remove_end_time)
 
-v1 = veto.indices_within_times(d.time1, veto_start, veto_end) 
-e = d.remove(v1)
+# Remove any zerolag triggers and get all exclusive background triggers
+exc_zero_trigs = all_trigs.remove(veto_indices1)
 
-v2 = veto.indices_within_times(e.time2, veto_start, veto_end) 
-e = e.remove(v2)
+# Veto indices from list of triggers for times in ifo 1 around the 
+# window gps times
+veto_indices2 = veto.indices_within_times(exc_zero_trigs.time2,
+                                          remove_start_time, remove_end_time)
+
+# Remove any coincident trigger pairs from the data set that could be in
+# coincidence with other coincident triggers (by the light travel time +
+# some padding indicated by veto-window) from ifo1 to average ifo.
+exc_zero_trigs = exc_zero_trigs.remove(veto_indices2)
 
 logging.info("Clustering coinc triggers (inclusive of zerolag)")
-d = d.cluster(args.cluster_window)
-fore_locs = d.timeslide_id == 0
+all_trigs = all_trigs.cluster(args.cluster_window)
+
+# Return an array of true or false if the trigger has not been
+# time-slid
+fore_locs = all_trigs.timeslide_id == 0
 logging.info("%s clustered foreground triggers" % fore_locs.sum())
 
 logging.info("Clustering coinc triggers (exclusive of zerolag)")
-e = e.cluster(args.cluster_window)
+exc_zero_trigs = exc_zero_trigs.cluster(args.cluster_window)
+
+# If a particular index of all_trigs.timeslide_id isn't 0, evaluate true.
+# So this returns an array of Booleans.
+back_locs = all_trigs.timeslide_id != 0
 
 logging.info("Dumping foreground triggers")
 f = fw(args.output_file)
-f.attrs['detector_1'] = d.attrs['detector_1']
-f.attrs['detector_2'] = d.attrs['detector_2']
-f.attrs['timeslide_interval'] = d.attrs['timeslide_interval']
+f.attrs['detector_1'] = all_trigs.attrs['detector_1']
+f.attrs['detector_2'] = all_trigs.attrs['detector_2']
+f.attrs['timeslide_interval'] = all_trigs.attrs['timeslide_interval']
 
+# Parse h_iterations to a string so we can use it later
+h_iter_string = str(h_iterations)
 # Copy over the segment for coincs and singles
-for key in d.seg.keys():
-    f['segments/%s/start' % key] = d.seg[key]['start'][:]
-    f['segments/%s/end' % key] = d.seg[key]['end'][:]
+for key in all_trigs.seg.keys():
```

Comment:
I did it this way in case in the future there are actually lots of triggers being hierarchically removed. If you feel this is better, I can revert it. I'm not sure what codes depend on this information downstream so this seemed like the most conservative way to do it.